src/iinit.cpp

```diff
@@ -433,6 +567,8 @@ void usage( char *prog ) {
     printf( "The client environment file is sought in ~/.irods/irods_environment.json\n" );
```

Comment:
sought... weird.   consider rewording this sentence for non-native english speakers.

Reply:
"sought" -> "found"?

Reply:
maybe instead... just change the sentence to be declarative of the order of operations...

```
Environment information is loaded from the following locations, with priority being given to the top of the list:
 - in specific environment variables
 - in an irods_environment.json file located at IRODS_ENVIRONMENT_FILE
 - in ~/.irods/irods_environment.json
 - default values set in the server
 ```

also, we should probably test to see whether that is all true...
if there is an IRODS_ENVIRONMENT_FILE defined
AND
say... some specific things also set... IRODS_DEFAULT_RESOURCE...

do they behave / get honored in the way described above?

Reply:
I'll try this out and report back.

Reply:
I'm not familiar with all of the environment variables, so I'll come back to testing that one. However, based on the code, for populating the client environment struct, it seems true that the environment variables take highest priority: https://github.com/irods/irods/blob/2979e35257c9e6454568371338a82862a3ea3718/lib/core/src/getRodsEnv.cpp#L121-L126 The environment file is consulted first, then indiscriminately overwrites values with those from environment variables (i.e. `getenv`) where they exist.

Confirmed that `IRODS_ENVIRONMENT_FILE` affects where `iinit` looks for the active client environment file (ignore the root-ness):
```bash
$ ls -l ~/.irods/
total 4
-rw-r--r-- 1 root root 169 Jun 26 18:34 irods_environment.json
$ IRODS_ENVIRONMENT_FILE=/lol.json iinit
 ERROR: environment_properties::capture: missing environment file. should be at [/lol.json]
Enter the host name (DNS) of the server to connect to: 
```
If not specified, defaults to looking in `~/.irods`, so that works as well.

Reply:
I updated the help text to read just like @trel suggested.

Tested out the environment variable thing and it worked out as expected. For this demonstration, the default resource is "demoResc" in the server:
```bash
$ cat ~/.irods/irods_environment.json  # no default resource in client environment
{
    "irods_host": "localhost",
    "irods_port": 1247,
    "irods_user_name": "alan",
    "irods_zone_name": "tempZone"
}
$ echo $IRODS_DEFAULT_RESOURCE     

$ iput putme foo1
$ cat ~/.irods/irods_environment.json # add a default resource to the client environment
{
    "irods_default_resource": "ufs0",
    "irods_host": "localhost",
    "irods_port": 1247,
    "irods_user_name": "alan",
    "irods_zone_name": "tempZone"
}
$ iput putme foo2
$ IRODS_DEFAULT_RESOURCE=ufs1 iput putme foo3
$ ils -l
/tempZone/home/alan:
  alan              0 demoResc          418 2023-06-26.19:11 & foo1
  alan              0 ufs0          418 2023-06-26.19:12 & foo2
  alan              0 ufs1          418 2023-06-26.19:12 & foo3
```

Reply:
Looks good to me.

Reply:
Yay, resolving